src/inspector_io.cc

```diff
@@ -175,15 +198,18 @@ bool InspectorIo::Start() {
     return false;
   }
   state_ = State::kAccepting;
-  if (options_.wait_for_connect()) {
+  if (wait_for_connect_) {
     DispatchMessages();
   }
   return true;
 }
 
 void InspectorIo::Stop() {
+  Write(TransportAction::kKill, 0, StringView());
```

Comment:
At this point the current state isn't known, `Start()` may not have been called, or may have failed, so there needs to be some state checks here before the _join.